qa/integration/services/service_container.rb

```diff
@@ -0,0 +1,77 @@
+require_relative "service"
+require "docker"
+
+# Represents a service running within a container.
+class ServiceContainer < Service
+
+  # Base image used by all service containers.
+  @@base_image = nil
```

Comment:
@wainersm let's not use any mutable `static/class` field for this:

1. (medium) Static global mutable is kind of dirty and could lead to a lot of bugs if the correct exec order is not met :)
2. (mayor) This creates a hard dependency between the Docker service and our tests. I think given that we have tiny Docker contexts to be build, it's fine to let Docker rebuild (which will effectively just mean hash the service directory contents in a few `ms`) on every service setup (mainly worried about the case of running this against Docker swarm here, which may or may not happen, let's stay a little flexible here :)).